server/models/Map.js

```diff
@@ -0,0 +1,207 @@
+const { Pool} = require('pg');
+const config = require('../config/config');
+const SQLCase2 = require("./sqls/SQLCase2");
+const SQLCase1 = require("./sqls/SQLCase1");
+const SQLCase3 = require("./sqls/SQLCase3");
+const SQLCase4 = require("./sqls/SQLCase4");
+const SQLZoomTargetCase1V2 = require("./sqls/SQLZoomTargetCase1V2");
+
+
+class Map{
+  constructor(){
+    this.pool = new Pool({ connectionString: config.connectionString });
+  }
+
+  async init(settings){
+    console.debug("init map with settings:", settings);
+    this.treeid = settings.treeid;
+    this.zoomLevel = parseInt(settings.zoom_level);
+    this.userid = settings.userid;
+    this.clusterRadius = settings.clusterRadius;
+    this.mapName = settings.map_name;
+    this.bounds = settings.bounds;
+    this.wallet = settings.wallet;
+    this.flavor = settings.flavor;
+    this.token = settings.token;
+    this.treeIds = [];
+    if(this.treeid){
```

Comment:
New version use simple, one-way if-else structure, it takes more lines of code, but easier to understand and maintain.